go/vt/vtctl/vtctl.go

```diff
@@ -3496,7 +3497,9 @@ func commandUpdateThrottlerConfig(ctx context.Context, wr *wrangler.Wrangler, su
 	customQuery := subFlags.String("custom-query", "", "custom throttler check query")
 	checkAsCheckSelf := subFlags.Bool("check-as-check-self", false, "/throttler/check requests behave as is /throttler/check-self was called")
 	checkAsCheckShard := subFlags.Bool("check-as-check-shard", false, "use standard behavior for /throttler/check requests")
-
+	throttledApp := subFlags.String("throttle-app", "", "an app name to throttle")
```

Comment:
How would a user know what valid app names exist? Without that, I think that this has limited value to the average user.

Can we add an exported variable or function here: https://github.com/vitessio/vitess/blob/main/go/vt/vttablet/tabletserver/throttle/throttlerapp/app.go#L29-L51

And then use that in the help output, e.g.:
```
throttledApp := subFlags.String("throttle-app", "", "an app name to throttle; valid values are %s", strings.Join(throttlerapp.GetNames(), ","))
```

Same for vtctldclient.

With that, we could also add some basic input validation so that a user is aware if they specified an invalid app name (e.g. mistyped `vstraemer`).

IMO these minor improvements would provide a lot of value.

Reply:
In the current design, app names are arbitrary. And while you have those well-known (how are they "well known" is a discussion of its own) apps like `online-ddl` or `vreplication`, there are also variants. Like, an Online DDL migration that uses `vreplication` answers both to `online-ddl` _and_ to `vreplication`, and _also_ it answers to the migration's UUID.

So, you can throttle all `online-ddl` operations ; or you can throttle all `vreplication` operations; or you can throttle a _specific migration_.

The list in https://github.com/vitessio/vitess/blob/1518cc10ca91120f3f402e75fb161862eb9cbaef/go/vt/vttablet/tabletserver/throttle/throttlerapp/app.go#L29-L51 is IMO pretty long to print out in a single CLI line.

I do see your point here, though! And I agree. But I don't have a good grasp of what would be the right solution. Most app names are irrelevant to the user and are really internal (e.g. I don't really want a user messing with `schema-tracker`, and if they do, that's on them). Maybe we should just let them know about the obvious ones?

The other thing is that we have a SQL interface to throttling migrations: `ALTER VITESS_MIGRATION [UUID] THROTTLE [ALL]`. In a followup PR I will make that SQL command go through `topo` (right now it goes directly to the tablets). So that's perhaps a formalization where the user doesn't even need to know the app name.

Open to discuss further.

Reply:
Ah, I see. Thanks! In that case then, I'd call them `built-in` / `vitess` / `internal` app names -- as we want Vitess operators to be able to use this with some ease. If you think it's a lot of work, we could also create a PR to make this improvement later (and think more about how best to do it). 

What do you think?

Reply:
We could also cover this in the website docs. If you think it's cumbersome / difficult in Vitess, we could at least do it in the docs and then that would be good enough IMO.

Reply:
Let's think about this a bit more. I think we need to understand better the usage patterns. For example, just the other day I've been asekd whether it's possible (it is with _this PR_) to only throttle `vcopier` but not `vplayer` as part of a VReplication / OnlineDDL workflow -- the reasoning for which was that they wanted to throttle for a duration of a few days, but did not want to break the workflow as binary logs were expected to rotate & purge.

Reply:
So I suggest let's document what we have for now, and follow up with another PR as needed.